src/Agent.Worker/ExecutionContext.cs

```diff
@@ -669,9 +669,13 @@ private string GetWorkspaceIdentifier(Pipelines.AgentJobRequestMessage message)
         // Do not add a format string overload. In general, execution context messages are user facing and
         // therefore should be localized. Use the Loc methods from the StringUtil class. The exception to
         // the rule is command messages - which should be crafted using strongly typed wrapper methods.
-        public long Write(string tag, string message)
+        public long Write(string tag, string message, bool canMaskSecrets = true)
         {
-            string msg = HostContext.SecretMasker.MaskSecrets($"{tag}{message}");
+            string msg = message;
```

Comment:
This can be simplified a bit
`string msg = canMaskSecrets ? HostContext.SecretMasker.MaskSecrets($"{tag}{message}") : message;`

Reply:
Agreed, thank you